### PR TITLE
Add key tasks and status field to pool modal

### LINKED ISSUE #2788: Add keyTasks and status fields to Pools model
# Description
`keyTasks` is just a longform text field which describes what kind of work this pool applies to. Can be expected to be many paragraphs long. This field should be optional

`status` should be a now PoolStatus enum, which currently will have only two values: TAKING_APPLICATIONS, NOT_TAKING_APPLICATIONS. This field should be required, but default to NOT_TAKING_APPLICATIONS.

## Acceptance Criteria

- [x] Add the enum to LocalizedConstants
- [x] Database migration to add to Pools table
- [x] add to Pool factories and seeders
- [x] Add these fields to edit and create forms in admin portal (also requires modifying grahql mutation inputs). Consider default & required value of status (keyTasks are optional, set status default to NOT_TAKING_APPLICATIONS).
- [x] Add fields to fakePool in storybook
- [x] Add fields to schema.graphql. Consider default & required value of status

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #2788 
- [x] Add the enum to LocalizedConstants
- [x] Database migration to add to Pools table
- [x] add to Pool factories and seeders
- [x] Add these fields to edit and create forms in admin portal (also requires modifying grahql mutation inputs). Consider default & required value of status (keyTasks are optional, set status default to NOT_TAKING_APPLICATIONS).
- [x] Add fields to fakePool in storybook
- [x] Add fields to schema.graphql. Consider default & required value of status

### CODE REVIEW

**Comment by vd1992** [comment]
Checkout -> refresh API -> seed
To start, it appears the model/migration/seed works. I can pull the new stuff in GraphQL-Playground

image

File: frontend/admin/src/js/components/pool/UpdatePool.tsx [hunk 1]
```diff
@@ -89,10 +93,6 @@ export const UpdatePoolForm: React.FunctionComponent<UpdatePoolFormProps> = ({
       en: values.name?.en,
       fr: values.name?.fr,
     },
-    description: {
-      en: values.description?.en,
-      fr: values.description?.fr,
-    },
```
Comment (vd1992) [inline frontend/admin/src/js/components/pool/UpdatePool.tsx:95]: Did you mean to remove this?

Comment (yonikid15) [reply frontend/admin/src/js/components/pool/UpdatePool.tsx]: I guess not 😆

**Comment by vd1992** [comment]
update operation now appears successful

image

File: frontend/common/src/fakeData/fakePools.ts [hunk 2]
```diff
@@ -36,6 +37,11 @@ const generatePool = (
     },
     classifications:
       faker.helpers.arrayElements<Classification>(classifications),
+    keyTasks: {
+      en: `EN ${faker.lorem.paragraph()}`,
+      fr: `FR ${faker.lorem.paragraph()}`,
+    },
+    status: faker.helpers.arrayElement<PoolStatus>(Object.values(PoolStatus)),
```
Comment (vd1992) [inline frontend/common/src/fakeData/fakePools.ts:44]: Might as well add this to the PoolStories too, in stories/pool.stories.tsx

Comment (yonikid15) [reply frontend/common/src/fakeData/fakePools.ts:44]: Done 👍

**Comment by yonikid15** [comment]
@petertgiles @tristan-orourke want to give this a final look over when you have a chance.

**Review by vd1992 — APPROVED** [review]
Think all is good as far as I can tell
Great work!

File: api/database/migrations/2022_05_30_182823_update_pools_table_with_key_tasks_and_status.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('pools', function (Blueprint $table) {
+            $table->jsonb('key_tasks')->nullable(false)->default(json_encode(['en' => '', 'fr' => '']));
+            $table->string('pool_status')->nullable(true);
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply api/database/migrations/2022_05_30_182823_update_pools_table_with_key_tasks_and_status.php:18]: Yup!

File: frontend/common/src/constants/localizedConstants.tsx [hunk 4]
```diff
@@ -643,3 +636,32 @@ export const getJobLookingStatusDescription = (
...
+): MessageDescriptor =>
+  getOrThrowError(
+    provinceOrTerritory,
+    provinceOrTerritoryId,
+    `Invalid province or territory '${provinceOrTerritoryId}'`,
+  );
+
+export const poolStatus = defineMessages({
+  [PoolStatus.NotTakingApplications]: {
+    defaultMessage: "NOT_TAKING_APPLICATIONS",
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply frontend/common/src/constants/localizedConstants.tsx:651]: There is no reason 😆

File: frontend/admin/src/js/components/pool/PoolTable.tsx [hunk 5]
```diff
@@ -75,6 +75,13 @@ export const PoolTable: React.FC<GetPoolsQuery & { editUrlRoot: string }> = ({
             );
           }),
       },
+      {
+        Header: intl.formatMessage({
+          defaultMessage: "Status",
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply frontend/admin/src/js/components/pool/PoolTable.tsx:80]: Axeing em

**Review by petertgiles — CHANGES_REQUESTED** [review]
(stupid github extension)

**Pull request merged**